src/tests/multihost/ipa/test_subid_ranges.py

```diff
@@ -0,0 +1,117 @@
+""" Automation of IPA bugs """
+
+import pytest
+import subprocess
+import time
+import os
+from sssd.testlib.common.utils import SSHClient
+
+
+test_password = "Secret123"
+user = 'admin'
+
+
+def execute_cmd(multihost, command):
+    """ Execute command on client """
+    cmd = multihost.client[0].run_command(command)
+    return cmd
+
+
+@pytest.mark.usefixtures('environment_setup',
+                         'subid_generate',
+                         'bkp_cnfig_for_subid_files',)
+@pytest.mark.tier1
+class TestSubid(object):
+    """
+    This is for ipa bugs automation
+    """
+    def test_subid_feature(self, multihost):
+        """
+        :Title: support subid ranges managed by FreeIPA
+        :id: 50bcdc28-00c8-11ec-bef4-845cf3eff344
+        :bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1803943
+        :steps:
+            1. Generate subid for user admin
+            2. Test newuidmap command
+            3. Test newgidmap command
+        :expectedresults:
+            1. Should succeed
+            2. Should succeed
+            3. Should succeed
+        """
+        ssh1 = SSHClient(multihost.client[0].ip,
+                         username=user, password=test_password)
+        (result, result1, exit_status) = ssh1.exec_command(f"ipa  "
+                                                           f"subid-find"
+                                                           f"  --owner  "
+                                                           f"{user}")
+        user_details = result1.readlines()
+        uid_start = int(user_details[5].split(': ')[1].split('\n')[0])
+        uid_range = int(user_details[6].split(': ')[1].split('\n')[0])
+        gid_start = int(user_details[7].split(': ')[1].split('\n')[0])
+        gid_range = int(user_details[8].split(': ')[1].split('\n')[0])
+        (results1, results2, results3) = ssh1.exec_command("unshare"
+                                                           " -U bash"
+                                                           " -c 'echo"
+                                                           " $$ > "
+                                                           "/tmp/unshare.pid;"
+                                                           " sleep 100'")
+        time.sleep(2)
+        proces_id = [int(i)
+                     for i in
+                     execute_cmd(multihost,
+                                 "cat /tmp/unshare.pid").stdout_text.split()
+                     if i.isdigit()][0]
+        uid = 0
+        count = 1
+        ssh1.exec_command(f"newuidmap {proces_id} {uid} {uid_start} {count}")
+        ssh1.exec_command(f"newgidmap {proces_id} {uid} {gid_start} {count}")
```

Comment:
Since `uid == 0` anyway, technically this is correct, but semantic (mapping `uid` to `gid_start`) is confusing.

Reply:
And once again: please check the return code of executed commands here.

Reply:
Running `newgidmap` with param named `uid` is onfusing ( https://github.com/SSSD/sssd/pull/5755#discussion_r704417721 )